File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const scores = {};
    name.forEach((n, i) => scores[n] = yearning[i]); // 객체로 만들기
    
    return photo.reduce((sum, eachPhoto) => {
       const eachSum = eachPhoto.reduce((s, n) => {
           if(Object.keys(scores).includes(n)) s += scores[n];
           return s;
       }, 0)
        sum.push(eachSum);
        return sum;
    }, [])
    
    
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
'''
dict로 namekey, yearning value
photo 2차원 순회하면서 점수 썸. 안찾아지면 pass
'''

def solution(name, yearning, photo):
    d = {}
    for i in range(len(name)):
        n = name[i]
        y = yearning[i]
        d[n] = y
        
    res = []
    for i in range(len(photo)):
        p = photo[i]
        s = 0
        for j in range(len(p)):
            person = p[j]
            
            if person in d:
                s += d[person]
        
        res.append(s)
            
    return res